sdk/spring/spring-cloud-azure-service/src/main/java/com/azure/spring/cloud/service/implementation/kafka/AzureOAuthBearerToken.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.service.implementation.kafka;
+
+import org.apache.kafka.common.security.oauthbearer.OAuthBearerToken;
+
+import java.time.Instant;
+import java.util.Set;
+
+/**
+ * Implementation of {@link OAuthBearerToken} for Azure Event Hubs.
+ */
+public class AzureOAuthBearerToken implements OAuthBearerToken {
+    private final String token;
+    private final Long startTimeMs;
+    private final long lifetimeMs;
+    private final Set<String> scope;
+    private final String principalName;
+
+    public AzureOAuthBearerToken(String token, Long startTimeMs, long lifetimeMs, Set<String> scope,
```

Comment:
Should this class accept an Azure core AccessToken?

Reply:
Resolved in #29404